src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -242,6 +253,87 @@ public bool NotUsingHttps(string url)
             }
         }
 
+        private async Task RecreateIndexes(DatabaseRecord databaseRecord)
+        {
+            PathSetting dataDirectory = databaseRecord.Settings.TryGetValue("DataDir", out var path) == false || path == null
```

Comment:
I think it would be better to use CreateDatabaseConfiguration from DatabasesLandlord (check note with IndexPath)

Reply:
FYI, never use configuration keys directly, use RavenConfiguration.GetKey(x => x.Core.DataDir) instead. It is not needed if the configuration will be created via DatabasesLandlord though, just informational.

Reply:
You don't have DataDir in the RavenConfiguration

Reply:
using CreateDatabaseConfiguration